test/servinge2e/verify_networkpolicy_test.go

```diff
@@ -48,8 +48,8 @@ func TestNetworkPolicy(t *testing.T) {
 
 	// Poll until network policy became active. It takes a few seconds.
 	err = wait.PollImmediate(test.Interval, 1*time.Minute, func() (bool, error) {
-		_, inErr := http.Get(ksvc.Status.URL.String())
-		if inErr == nil {
+		resp, inErr := http.Get(ksvc.Status.URL.String())
+		if inErr == nil || resp.StatusCode == http.StatusOK {
```

Comment:
Ah, sorry my bad. When `http.StatusBadGateway` was returned, `inErr` is `nil`. So we need to change like this:

```go
                if resp != nil && resp.StatusCode == http.StatusOK {
                        t.Logf("Network policy did not block the request to %s", ksvc.Status.URL.String())
                        return false, nil
                }
                if inErr == nil {
                        t.Logf("Network policy did not block the request to %s", ksvc.Status.URL.String())
                        return false, nil
                }
```

Reply:
I believe `if inErr == nil && resp.StatusCode == http.StatusOK {` expresses the same. Pushed.